### PR TITLE
ON-41352 # Changed secret to organisationManagedSecretId in NewFormApprovalWebhook

### PR DESCRIPTION
**Pull request by mymattcarroll**

### CODE REVIEW

File: typescript/approvals.d.ts [hunk 1]
```diff
@@ -189,16 +189,32 @@ export type FormApprovalWebhookEvent =
...
+  /**
+   * `secret` is optional when updating a webhook. If no secret is passed, the
+   * secret will remain unchanged.
+   */
+  secret?: string
+}
+
+export type NewFormApprovalWebhook = BaseFormApprovalWebhook & {
+  customerSecretId: number
+}
```
Comment (Zaxist) [inline typescript/approvals.d.ts:216]: This seems really confusing to me. Why does a `NewFormApprovalWebhook` require an existing secretId?

Comment (mymattcarroll) [reply typescript/approvals.d.ts:216]: `NewFormApprovalWebhook` is the type passed to our database layer. It requires everything that is needed to create a `FormApprovalWebhook` from the database layer's perspective. That is the pattern the majority (potentially all) of our `New*` types follow.

Comment (Zaxist) [reply typescript/approvals.d.ts:216]: Hm. For most it is also the type the client needs to pass to the api though.

Comment (Zaxist) [reply typescript/approvals.d.ts:216]: I understand it seems obvious to you, but you are one person who thinks about it in one way.

Comment (mymattcarroll) [reply typescript/approvals.d.ts:216]: Happy coincidence. I'm happy to rename the type: `FormApprovalWebhookDatabaseCreatePayload`?

File: typescript/approvals.d.ts [hunk 2]
```diff
@@ -189,22 +189,34 @@ export type FormApprovalWebhookEvent =
...
+export type FormApprovalWebhookUpdatePayload = BaseFormApprovalWebhook & {
+  /**
+   * `secret` is optional when updating a webhook. If no secret is passed, the
+   * secret will remain unchanged.
+   */
+  secret?: string
 }
 
+export type NewFormApprovalWebhook = BaseFormApprovalWebhook & {
+  organisationManagerSecretId: number
```
Comment (Zaxist) [inline typescript/approvals.d.ts:215]:
```suggestion
  organisationManagedSecretId: number
```

File: typescript/misc.d.ts [hunk 3]
```diff
@@ -189,15 +189,22 @@ export type EndpointConfiguration =
...
+  /** The ARN of the KMS key alias that was used to encrypt the secret */
   kmsKeyAliasArn: string
+  /** The ARN of the KMS key that was used to encrypt the secret */
   kmsKeyArn: string
 }
 
+export type OrganisationManagedSecret = CustomerSecret & {
+  /** Display text to represent the secret */
+  label: string
+  /** The identifier for the organisation associated with the Lookup. */
```
Comment (mymattcarroll) [inline typescript/misc.d.ts:204]:
```suggestion
  /** The identifier for the organisation associated with the secret. */
```

**Pull request merged**